### ISSUE TITLE
WCAG 2.0 (Level AA) - Gantry offcanvas toggle

### ISSUE THREAD

**Issue by marktaylor46** [gantry/gantry5#1219]
The use of the <i> (italic) tab for the offfcanvas should be eliminated. I suggest that it is change to a SPAN tag instead.
The reason for this is that it is reported as an error in Accessibility checkers e.g. (http://achecker.ca/checker/index.php).

The file affected is /media/gantry5/engines/nucleus/templates/page.html.twig line 69

![2016-04-01_10-14-49](https://cloud.githubusercontent.com/assets/11683218/14202913/a98aebcc-f7f2-11e5-96d4-dc093cd7862f.png)

We should stop using the <i> (italic) tag anywhere else in Gantry 5 or templates too.

**Comment by mahagr** [comment]
Italic tag is used by an icon and we cannot remove it or there would not be anything to click on.

**Comment by marktaylor46** [comment]
huh? I tried changing it to SPAN instead and it works absolutely fine? The icon is actually coming from the class not the html tag?

**Comment by mahagr** [comment]
There is no text in the tag. Its not an accessibility issue as there's no text to read.

**Comment by marktaylor46** [comment]
I still don't see why the tag used cannot be changed. Why let checkers report this as an error unnecessarily when we have gone to so much trouble already to be WCAG2.0 compliant?

**Comment by mahagr** [comment]
https://fortawesome.github.io/Font-Awesome/examples/ <- this is the recommended way to use icons.

**Comment by JoomFX** [comment]
You can use `<span>` instead of `<i>` with the same result. It doesn't really matter.
Even though the specification of `<i>` says that it is for _Italian text_ it has become almost as a standard to use the tag for icon fonts. So I don't think you should bother to change the `<i>` tag to `<span>` even if it is reported as an issue by some tools.

**Comment by marktaylor46** [comment]
I understand what fontawesome says but it is technically not compliant with WCAG specification. So the error being reported is technically correct, even though I agree in reality it makes no difference. But... (and this is the main point) not everyone is aware of this and will see Gantry 5 as not being WCAG2.0 compliant. So, for me, for the very little effort it takes it seems an easy win for me to avoid questions and misunderstandings.

**Comment by mahagr** [comment]
I agree with @JoomFX. The tool is reporting false positive and its an issue in the tool itself, not in the site. There is NO accessibility issue as there's nothing for user to see.

**Comment by marktaylor46** [comment]
Ok... I tried :)

**Comment by Bokelmann** [comment]
I say: Make it accessible as possible. If <i> is bad and unsemantic bad... you should change it.

**Comment by mahagr** [comment]
If you say that `<i>` is bad and unsemantic, so is most of the markup, including `<span>` and `<div>`. There's nothing bad in using `<i>` unless you use it inside a paragraph to emphasis something. As it has no semantic meaning, it can only been seen as italic (or in this case as an icon), not heard or felt. This means that `<i>` tag is actually great on using on something that is just a visual detail in the page -- for example as an icon.

This was probably the reason why `<em>` and `<strong>` was added to html spec instead of just adding the semantic meaning to `<i>` and `<b>` tags.

If there's nothing broken, do not try to fix it. I cannot do anything to prevent users from using accessibility checkers that are 5 years old (2011 in this case) and do not support the new markup standards.

**Comment by marktaylor46** [comment]
I have checked it with a more modern check http://wave.webaim.org/ and that does seem to say it's ok so I guess we're alright.

**Comment by Bokelmann** [comment]
> If you say that ` <i> ` is bad and unsemantic

I did't say that ... I said IF ... don't to jump at conclusions so quickly.
If it's alright from the point of accessibility .... I'm totally fine.
Seems like your are an expert on accessibility ... I'm not.

**Comment by mahagr** [comment]
@Bokelmann Sorry, I didn't mean to put words into your mouth but just explain why I didn't think its an issue.
